### DISCUSSION TITLE
How to check for multiple keys in themes?

### DISCUSSION THREAD

**Discussion by DIrgi** [mapsforge/vtm#1120] [Q&A] [answered]
Hi,
I'm just trying to create my own theme for mapforge. Analysing existing themes are very helpful for that.
However at the moment I'm stucked in finding a solution or the "how to" when it comes to analyze more than one key.
I found many solutions for one key having several values but having several keys.....

![image](https://github.com/mapsforge/mapsforge/assets/57968526/8d028c85-fac7-4a4e-ba86-2f870a7d8b75)

In the picture you see the key data existing for a track in OSM. 
I want to check for highway, motor_vehicle and finally tracktype at the same time so when:

> **Highway= track** and **motor_vehicle=yes** and **tracktype=gradeX** then.....

Does mapsforge support this and how has the xml rule to llike then?

Maybe someone has a tip for me.
Thank you very much.

**Comment by devemux86** [comment, marked as answer]
@DIrgi 

You can use nested rules to combine checks on multiple key=value.
For example, see the highway rules in the themes, they are structured that way.

---

By the way, are you sure you are in the right forum? This is for [Mapsforge themes](https://github.com/mapsforge/mapsforge/tree/master/mapsforge-themes/src/main/resources/assets/mapsforge).
If you work on [VTM (OpenGL) themes](https://github.com/mapsforge/vtm/tree/master/vtm-themes/resources/assets/vtm),  then please use this repository: https://github.com/mapsforge/vtm

**Reply by DIrgi** [reply]
Thanks for your reply.
Ok, I post it there - Sorry was not really clear for me what is what

**Reply by devemux86** [reply]
@DIrgi I transferred the discussion in VTM repository.

**Reply by DIrgi** [reply]
You where faster then me I just established a new one on the right place - so better you should delete them both.

**Reply by devemux86** [reply]
@DIrgi however they are different questions.
